object-manipulation-2/src/pick.js

```diff
@@ -2,11 +2,14 @@
 function pick(source, keys) {
   var object = {};
   for (var i = 0; i < keys.length; i++) {
-    for (var key in source) {
-      if (key === keys[i] && source[key] !== undefined) {
-        object[key] = source[key];
+    if (source[keys[i]] !== undefined) {
+      for (var key in source) {
+        if (key === keys[i]) {
+          object[key] = source[key];
+        }
       }
     }
```

Comment:
I'm pretty sure you can just change this to this:
```
    if (source[keys[i]] !== undefined) {
        var key = keys[i];
        object[key] = source[key];
    }
```

Reply:
This removes an extra conditional and loop, which keeps your code cleaner and the goal of the code clear.